source/loader/CMakeLists.txt

```diff
@@ -48,6 +48,13 @@ if (UNIX)
     set(THREADS_PREFER_PTHREAD_FLAG TRUE)
     find_package(Threads REQUIRED)
     target_link_libraries(ur_loader PRIVATE Threads::Threads)
+    # Older gcc versions need -pthread, not just -lpthread
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11.1)
+            find_library(PTHREAD_LIBRARY pthread)
```

Comment:
The result of this `find_library` is not used, if you wanted to assert that `pthread` is available on the system then `PTHREAD_LIBRARY`'s value should be checked to ensure it does not contain `NOTFOUND` or add the `REQUIRED` flag to the `find_library` call. Alternatively, we could assume a POSIX system when using GCC and completely remove this `find_library` call.

Reply:
I've changed so that we are checking for `CMAKE_USE_PTHREADS_INIT` which is set by cmake `Threads`.